sources/mongodb/__init__.py

```diff
@@ -0,0 +1,85 @@
+"""Source that loads collections form any a mongo database, supports incremental loads."""
+
+from typing import List, Optional, Iterable, Any
+
+import dlt
+from dlt.extract.source import DltResource
+
+from .helpers import (
+    collection_documents,
+    client_from_credentials,
+    MongoDbCollectionConfiguration,
+    MongoDbCollectionResourceConfiguration,
+)
+
+@dlt.source
+def mongodb(
+    connection_url: str = dlt.secrets.value,
+    database: str = dlt.config.value,
+    collection_names: Optional[List[str]] = dlt.config.value,
+    incremental: Optional[dlt.sources.incremental[Any]] = dlt.config.value,
+) -> Iterable[DltResource]:
+    """
+    A DLT source which loads data from a mongo database using PyMongo.
+    Resources are automatically created for each collection in the database or from the given list of collection.
+
+    Args:
+        connection_url (str): Database connection_url.
+        database (str): Selected database name.
+        collection (Optional[List[str]]): The list of collections `pymongo.collection.Collection` to load.
+        incremental (Optional[dlt.sources.incremental[Any]]): Option to enable incremental loading for the collection.
+            E.g., `incremental=dlt.sources.incremental('updated_at', pendulum.parse('2022-01-01T00:00:00Z'))`
+
+    Returns:
+        Iterable[DltResource]: A list of DLT resources for each collection to be loaded.
+    """
+
+    # set up mongo client
+    client = client_from_credentials(connection_url)
+    database = client[database]
+
+    # use provided collection or all conllections
+    if not collection_names:
+        collection_names = database.list_collection_names()
+
+    collection_list = [database[collection] for collection in collection_names]
+
+    for collection in collection_list:
+        yield dlt.resource(
+            collection_documents,
+            name=collection.name,
+            primary_key="_id",
+            spec=MongoDbCollectionConfiguration,
+        )(client, collection, incremental=incremental)
+
+
+@dlt.common.configuration.with_config(
+    sections=("sources", "mongodb"), spec=MongoDbCollectionResourceConfiguration
+)
+def mongodb_collection(
+    connection_url: str = dlt.secrets.value,
+    database: str = dlt.config.value,
+    collection: str = dlt.config.value,
```

Comment:
we should also add `write_disposition` here. if `sql_database` does not have it we'll add it as well